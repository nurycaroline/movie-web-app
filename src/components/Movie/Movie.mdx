---
name: Movie
menu: Componentes
---

import {Playground} from 'docz';
import Movie from '.';

# Movie

<Playground>
  <Movie
    title="John Wick 3 - Parabellum"
    image={"/cBO4u9qR1A7kSU299Qr3Ru9lvZ3.jpg"}
    overview="Após assassinar o chefe da máfia Santino D'Antonio (Riccardo Scamarcio) no Hotel Continental, John Wick (Keanu Reeves) passa a ser perseguido pelos membros da Alta Cúpula sob a recompensa de U$14 milhões. Agora, ele precisa unir forças com antigos parceiros que o ajudaram no passado enquanto luta por sua sobrevivência."
    genres="Crime,Ação,Thriller"
  />
</Playground>
