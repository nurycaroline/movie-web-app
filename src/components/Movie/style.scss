@import '../../assets/variables';

.movie {
  max-width: 500px;
  width: 100%;
  display: flex;
  flex-direction: row;
  min-height: 300px;
  margin-bottom: 15px;
  box-shadow: 1px 5px 5px $shadown-color;

  .information {
    min-width: 200px;
    padding: 0 15px;

    .title {
      color: $second-color;
      font-weight: bold;
      font-size: 18px;
      padding: 15px;
    }
    .overview {
      overflow: hidden;
      min-width: 200px;
      height: 160px;

      display: -webkit-box;
      -webkit-line-clamp: 6;
      -webkit-box-orient: vertical;
      overflow: hidden;
      text-overflow: ellipsis;
      line-height: 1.625;
    }
    .genres {
      display: flex;
      justify-content: flex-end;
      align-items: end;
      height: 50px;
      flex-direction: column;
      line-height: 20px;
    }
  }
}
