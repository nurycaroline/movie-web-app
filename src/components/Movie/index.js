import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

const Movie = ({
  title, image, overview, genres,
}) => (
  <div className="movie">
    {image && (
      <img src={`https://image.tmdb.org/t/p/w200${image}`} alt="" />
    )}
    <div className="information">
      <p className="title">{title}</p>
      <p className="overview">{overview}</p>
      <p className="genres">{genres}</p>
    </div>
  </div>
);

Movie.propTypes = {
  title: PropTypes.string,
  image: PropTypes.string,
  overview: PropTypes.string,
  genres: PropTypes.string,
};

Movie.defaultProps = {
  title: '',
  image: '',
  overview: '',
  genres: '',
};

export default Movie;
