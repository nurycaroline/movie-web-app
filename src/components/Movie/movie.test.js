import React from 'react';
import renderer from 'react-test-renderer';
import Movie from '.';

test('Movie', () => {
  const component = renderer.create(
    <Movie
      title="John Wick 3 - Parabellum"
      image="/cBO4u9qR1A7kSU299Qr3Ru9lvZ3.jpg"
      overview="Após assassinar o chefe da máfia Santino D'Antonio (Riccardo Scamarcio) no Hotel Continental, John Wick (Keanu Reeves) passa a ser perseguido pelos membros da Alta Cúpula sob a recompensa de U$14 milhões. Agora, ele precisa unir forças com antigos parceiros que o ajudaram no passado enquanto luta por sua sobrevivência."
      genres="Crime,Ação,Thriller"
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
