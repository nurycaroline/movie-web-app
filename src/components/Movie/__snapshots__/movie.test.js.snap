// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie 1`] = `
<div
  className="movie"
>
  <img
    alt=""
    src="https://image.tmdb.org/t/p/w200/cBO4u9qR1A7kSU299Qr3Ru9lvZ3.jpg"
  />
  <div
    className="information"
  >
    <p
      className="title"
    >
      John Wick 3 - Parabellum
    </p>
    <p
      className="overview"
    >
      Após assassinar o chefe da máfia Santino D'Antonio (Riccardo Scamarcio) no Hotel Continental, John Wick (Keanu Reeves) passa a ser perseguido pelos membros da Alta Cúpula sob a recompensa de U$14 milhões. Agora, ele precisa unir forças com antigos parceiros que o ajudaram no passado enquanto luta por sua sobrevivência.
    </p>
    <p
      className="genres"
    >
      Crime,Ação,Thriller
    </p>
  </div>
</div>
`;
